ftp-utils/src/main/java/org/wso2/ei/b7a/ftp/core/util/FTPUtil.java

```diff
@@ -158,4 +167,49 @@ public static BType getFileInfoType() {
                 FTPConstants.FTP_FILE_INFO);
         return fileInfoStruct.getType();
     }
+
+    public static ByteArrayInputStream compress(InputStream inputStream, String targetFilePath) {
+
+        String fileName = new File(targetFilePath).getName();
+
+        try {
+
+            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+            ZipOutputStream zipOutputStream = new ZipOutputStream(byteArrayOutputStream);
+            zipOutputStream.putNextEntry(new ZipEntry(fileName));
+            BufferedInputStream bufferedInputStream = new BufferedInputStream(inputStream);
+
+            byte[] buffer = new byte[1000];
+            int len;
+            while ((len = bufferedInputStream.read(buffer)) > 0) {
+                zipOutputStream.write(buffer, 0, len);
+            }
+            bufferedInputStream.close();
+            zipOutputStream.closeEntry();
+            zipOutputStream.close();
```

Comment:
Shouldn't this be done in a finally block?

Reply:
Fixed.